sycl/include/CL/sycl/device.hpp

```diff
@@ -28,13 +28,19 @@ class device_impl;
 auto getDeviceComparisonLambda();
 }
 
+// A tag to allow for creating a host device
+struct host_device {};
+
 /// The SYCL device class encapsulates a single SYCL device on which kernels
 /// may be executed.
 ///
 /// \ingroup sycl_api
 class __SYCL_EXPORT device {
 public:
   /// Constructs a SYCL device instance as a host device.
+  device(host_device);
```

Comment:
I don't see this constructor in the specification. Why do you think we should add it?

Reply:
I don't think there's an approval to remove host device completely. Also, libsycl.so makes use of host device.

Reply:
I'm not suggesting we move the host device support completely, but we should not add silently add non-standard functionality w/o extending the specification.

You can use `aspect_selector` with `host` aspect to select a host device. The host aspect is not in the standard, but it's already supported by DPC++, so no new extensions are needed.